# HG changeset patch
# Parent cef1817c3b13bcb5065b721a35dee94a2940e339
# User Frédéric Wang <fred.wang@free.fr>
munderover: consider munder vertical metrics during the munder attachment phase instead of the mover attachment phase.

diff --git a/layout/mathml/nsMathMLmunderoverFrame.cpp b/layout/mathml/nsMathMLmunderoverFrame.cpp
--- a/layout/mathml/nsMathMLmunderoverFrame.cpp
+++ b/layout/mathml/nsMathMLmunderoverFrame.cpp
@@ -414,18 +414,17 @@ nsMathMLmunderoverFrame::Place(nsRenderi
   else {
     mBoundingMetrics.width = NS_MAX(bmBase.width, overWidth);
     dxOver += correction/2 + (mBoundingMetrics.width - overWidth)/2;
   }
   dxBase = (mBoundingMetrics.width - bmBase.width)/2;
 
   mBoundingMetrics.ascent = 
     bmBase.ascent + overDelta1 + bmOver.ascent + bmOver.descent;
-  mBoundingMetrics.descent = 
-    bmBase.descent + underDelta1 + bmUnder.ascent + bmUnder.descent;
+  mBoundingMetrics.descent = bmBase.descent;
   mBoundingMetrics.leftBearing = 
     NS_MIN(dxBase + bmBase.leftBearing, dxOver + bmOver.leftBearing);
   mBoundingMetrics.rightBearing = 
     NS_MAX(dxBase + bmBase.rightBearing, dxOver + bmOver.rightBearing);
 
   //////////
   // pass 2, do what <munder> does: attach the underscript on the previous
   // result. We conceptually view the previous result as an "anynomous base" 
@@ -458,16 +457,19 @@ nsMathMLmunderoverFrame::Place(nsRenderi
 
   // adjust the offsets of the real base and overscript since their
   // final offsets should be relative to us...
   dxOver += dxAnonymousBase;
   dxBase += dxAnonymousBase;
 
   mBoundingMetrics.width =
     NS_MAX(dxAnonymousBase + bmAnonymousBase.width, dxUnder + bmUnder.width);
+  // At this point, mBoundingMetrics.ascent = bmAnonymousBase.ascent 
+  mBoundingMetrics.descent = 
+    bmAnonymousBase.descent + underDelta1 + bmUnder.ascent + bmUnder.descent;
   mBoundingMetrics.leftBearing =
     NS_MIN(dxAnonymousBase + bmAnonymousBase.leftBearing, dxUnder + bmUnder.leftBearing);
   mBoundingMetrics.rightBearing = 
     NS_MAX(dxAnonymousBase + bmAnonymousBase.rightBearing, dxUnder + bmUnder.rightBearing);
 
   aDesiredSize.ascent = ascentAnonymousBase;
   aDesiredSize.height = aDesiredSize.ascent +
     NS_MAX(mBoundingMetrics.descent + underDelta2,
